File: bindings/nostr-sdk-nodejs/examples/client_with_opts.js
const { Keys, Client, Options, initLogger } = require("../index");

async function main() {
    initLogger();

    let keys = Keys.fromSkStr("nsec1ufnus6pju578ste3v90xd5m2decpuzpql2295m3sknqcjzyys9ls0qlc85");
    let opts = new Options().waitForConnection(true).waitForSend(true);

    let client = Client.newWithOpts(keys, opts);
    await client.addRelay("wss://relay.damus.io");
    await client.addRelay("wss://nostr.oxtr.dev");
    await client.addRelay("wss://nostr.bitcoiner.social");
    await client.addRelay("wss://nostr.openchain.fr");
    await client.addRelay("ws://jgqaglhautb4k6e6i2g34jakxiemqp6z4wynlirltuukgkft2xuglmqd.onion", "127.0.0.1:9050");

    await client.connect();

    await client.publishTextNote("Hello World!", []);
}

main();

File: bindings/nostr-sdk-ffi/bindings-python/examples/client-with-opts.py
from nostr_sdk import Keys, Client, Options, EventBuilder, Proxy, ProxyTarget, init_logger, LogLevel
from datetime import timedelta

init_logger(LogLevel.INFO)

keys = Keys.generate()
print(keys.public_key().to_bech32())

# Configure client to use proxy for `.onion` relays
proxy = Proxy("127.0.0.1:9050").target(ProxyTarget.ONION)
opts = (Options()
        .connection_timeout(timedelta(seconds=60))
        .send_timeout(timedelta(seconds=10))
        .proxy(proxy))
client = Client.with_opts(None, opts)

client.add_relays([
    "wss://relay.damus.io",
    "ws://oxtrdevav64z64yb7x6rjg4ntzqjhedm5b5zjqulugknhzr46ny2qbad.onion"
])
client.connect()

event = EventBuilder.text_note("Hello from Rust Nostr Python bindings!", []).to_event(keys)
event_id = client.send_event(event)
print("Event sent:")
print(f" hex:    {event_id.to_hex()}")
print(f" bech32: {event_id.to_bech32()}")
